src/compiler/transformers/es2015.ts

```diff
@@ -1146,7 +1146,7 @@ namespace ts {
                         [
                             ...existingPrologue,
                             ...prologue,
-                            ...(superStatementIndex <= existingPrologue.length ? emptyArray : visitNodes(constructor.body.statements, visitor, isStatement, existingPrologue.length, superStatementIndex)),
+                            ...(superStatementIndex <= existingPrologue.length ? emptyArray : visitNodes(constructor.body.statements, visitor, isStatement, existingPrologue.length, superStatementIndex - existingPrologue.length)),

```

Comment:
The last argument is `count` and not `end`, so I believe the fix is to substract the start position from here. Otherwise we're processing too many nodes (including `super()`)

Reply:
Good catch!